src/gd_avif.c

```diff
@@ -361,7 +361,7 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFromAvifCtx (gdIOCtx *ctx)
 	// items. libheif v1.11.0 or older does not add the 'pixi' item property to
 	// AV1 image items. (This issue has been corrected in libheif v1.12.0.)
 
-#if AVIF_VERSION_MAJOR > 0 || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR > 9) || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR == 9 && AVIF_VERSION_PATCH >= 1)
+#if AVIF_VERSION >= 90100
```

Comment:
The way major, minor, and patch versions are packed into AVIF_VERSION changed after libavif v0.8.3.

v0.8.3:
#define AVIF_VERSION (AVIF_VERSION_MAJOR * 10000) + (AVIF_VERSION_MINOR * 100) + AVIF_VERSION_PATCH

v0.8.4:
#define AVIF_VERSION \
    ((AVIF_VERSION_MAJOR * 1000000) + (AVIF_VERSION_MINOR * 10000) + (AVIF_VERSION_PATCH * 100) + AVIF_VERSION_DEVEL)

This is unlikely to change again, and I believe the new test should still work:

v0.8.3: AVIF_VERSION = 803
v0.8.4: AVIF_VERSION = 80400

Reply:
I guess we're just lucky here. Which doesn't feel great. But I'm quite happy to be able to use the more legible form `AVIF_VERSION` affords us.

I'm going to push this into https://github.com/php/php-src/pull/7253 as well.